packages/core/notifications/core-notifications-browser/src/types.ts

```diff
@@ -18,8 +19,8 @@ import type { MountPoint } from '@kbn/core-mount-utils-browser';
  * @public
  */
 export type ToastInputFields = Pick<EuiToast, Exclude<keyof EuiToast, 'id' | 'text' | 'title'>> & {
-  title?: string | MountPoint;
-  text?: string | MountPoint;
+  title?: string | MountPoint | ReactNode;
+  text?: string | MountPoint | ReactChild;
```

Comment:
The intend here was (is) to keep the API rendering-framework agnostic (as all other core APIs).

What do you need to do with a plain `ReactNode` or `ReactChild` child you can't do with a `MountPoint`?

Reply:
We wanted to allow for the (title+notification badge) combo in the collapsed titles. For that, we need a non-string/react component to be rendered. When using `MountPoint`s with `mountReactNode()`, we'll start getting react warnings because we try to call unmount on elements. When just rendering plain react nodes, this warning didn't show up - would this be a bad extension? The API is agnostic, in the sense that you can still use any mount mechanism, but you can also use react for convenience - or would that introduce unwanted dependencies?

Reply:
Is it not advised to extend this API with react flavor? It still can be used in an agnostic way. 

This is the error we're getting if I'm using `title: mountReactNode(...)`:
```
Warning: unmountComponentAtNode(): The node you're attempting to unmount was rendered by React and is not a top-level container. Instead, have the parent component update its state and rerender in order to remove this component.
``` 
I'm not exactly sure what this refers to

Reply:
> Is it not advised to extend this API with react flavor?

Unfortunately it is not, no. Exposing react APIs / leaking react types from Core public API is not allowed. 

Internally using react as an implementation detail (or for Core *internal* types) is perfectly fine though. 

So the correct way would probably be to introduce a internal type having the changes you made on `ToastInputFields` and uses this type internally, while still having the public type / API using the unchanged `ToastInputFields`.

That way you can use react nodes/elements as `title`/`body` internally for the changes needed for this PR without changing the shape of the public API we're exposing.

Reply:
I understand. I'll try to adopt some changes not to expose React on the API. I'll ping back soon.

Reply:
Done in: https://github.com/elastic/kibana/pull/161738/commits/dbed104f9c74c6f78f62c541ee290cdd5948b372